File: Mobile/backend/src/main/java/com/example/hsbcproject/dto/StockCandleResponse.java
package com.example.hsbcproject.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

/**
 * Public response DTO for 30-day (or N-day) OHLCV stock history.
 * Used by GET /api/prices/history/{ticker}
 */
public record StockCandleResponse(
        String ticker,
        String source,          // "FINNHUB" or "DUMMY"
        List<DailyCandle> candles) {

    public record DailyCandle(
            LocalDate date,
            BigDecimal open,
            BigDecimal high,
            BigDecimal low,
            BigDecimal close,
            BigDecimal volume) {
    }
}

File: Mobile/app/src/main/java/com/hsbc/portfoliomanager/data/model/StockCandleResponse.kt
package com.hsbc.portfoliomanager.data.model

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

/** Mirrors backend StockCandleResponse — returned by GET /api/prices/history/{ticker} */
data class StockCandleResponse(
    @SerializedName("ticker")
    val ticker: String,

    /** "FINNHUB" for stocks, "DUMMY" for bonds/crypto */
    @SerializedName("source")
    val source: String,

    @SerializedName("candles")
    val candles: List<DailyCandle>
)

data class DailyCandle(
    @SerializedName("date")
    val date: String,   // "YYYY-MM-DD"

    @SerializedName("open")
    val open: BigDecimal,

    @SerializedName("high")
    val high: BigDecimal,

    @SerializedName("low")
    val low: BigDecimal,

    @SerializedName("close")
    val close: BigDecimal,

    @SerializedName("volume")
    val volume: BigDecimal
)
